llama.cpp

```diff
@@ -15849,6 +15853,18 @@ struct llama_context * llama_new_context_with_model(
             }
             ctx->backends.push_back(backend);
         }
+#elif defined(GGML_USE_QNN)
+        if (model->n_gpu_layers > 0) {
+            //the second param is data path of prebuit QNN libs provided by Qualcomm
+            //can be hardcoded to "/data/local/tmp/"
+            ggml_backend_t backend = ggml_backend_qnn_init(model->main_gpu, "/data/local/tmp/");
```

Comment:
I'm recommend let system auto find so files with LD_LIBRARY_PATH or PATH. (onnxruntime's QNN also has this behavior.

Reply:
> I'm recommend let system auto find so files with LD_LIBRARY_PATH or PATH. (onnxruntime's QNN also has this behavior.

thanks for your comments. I'll try and investigate implementation of onnxruntime's QNN.

Reply:
My implement is setup `PATH` env.

https://github.com/mybigday/llama.node/blob/575219e4c51848c8d12abfcd9c691b533052b837/lib/binding.ts#L55-L66

`dlopen("libQnnXXX.so")` or `LoadLibrary("QnnXXX.dll")` will auto find them.

P.S. For Android apk lib path should also in auto find path.

Reply:
I think this method might be not properly on Android, pls refer to:https://github.com/zhouwg/kantv/blob/master/cdeosplayer/kantv/src/main/java/com/cdeos/kantv/app/IApplication.java#L253